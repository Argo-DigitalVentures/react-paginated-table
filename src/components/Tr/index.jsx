import React, { PropTypes } from 'react';

import {
  dataType,
  columnType,
  classNameType,
  styleType,
} from '../../propTypes';

import { evaluate } from '../../utils';

function Tr(props) {
  const {
    columns,
    row,
    rowIndex,
    data,
    style: rowStyle,
    className: rowClassName,
    CellComponent,
    cellStyle,
    cellClassName,
    ...rest
  } = props;
  const dataProps = { columns, row, rowIndex, data };
  const className = evaluate(rowClassName, dataProps);
  const style = evaluate(rowStyle, dataProps);
  return (
    <tr style={style} className={className} {...rest}>
      {
        columns
          .filter(({ hidden }) => !hidden)
          .map((column, columnIndex, columns) => (
            <CellComponent
              key={column.id}
              style={cellStyle}
              className={cellClassName}
              column={column}
              columnIndex={columnIndex}
              columns={columns}
              row={row}
              rowIndex={rowIndex}
              data={data}
            />
          ))
      }
    </tr>
  );
}

Tr.defaultProps = {
  style: {},
  className: '',
  cellStyle: {},
  cellClassName: '',
};

Tr.propTypes = {
  data: dataType.isRequired,
  columns: PropTypes.arrayOf(columnType).isRequired,
  row: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  rowIndex: PropTypes.number.isRequired,
  CellComponent: PropTypes.func.isRequired,
  style: styleType,
  className: classNameType,
  cellStyle: styleType,
  cellClassName: classNameType,
};

export default Tr;
