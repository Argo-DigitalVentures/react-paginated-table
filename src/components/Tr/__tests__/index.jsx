import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Tr from '../';

const data = [
  { company: 'Argo', location: 'New York' },
  { company: 'John Hancock', location: 'New York' },
];

const columns = [
  { id: 'company', header: 'Company Name', selector: 'company' },
  { id: 'location', header: 'Location', selector: 'location' },
];

const FakeCellComponent = () => null;

describe('<Tr />', () => {
  it('should accept className, style, and other DOM props', () => {
    const tree = shallow(
      <Tr
        className="test"
        style={{
          color: 'green',
        }}
        title="This is a row in the table."
        data={data}
        columns={columns}
        row={data[1]}
        rowIndex={1}
        CellComponent={FakeCellComponent}
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });

  it('should accept className and style functions', () => {
    const classNameFn = jest.fn().mockReturnValue('test');
    const styleFn = jest.fn().mockReturnValue({
      color: 'green',
    });
    const tree = shallow(
      <Tr
        className={classNameFn}
        style={styleFn}
        title="This is a row in the table."
        data={data}
        columns={columns}
        row={data[1]}
        rowIndex={1}
        CellComponent={FakeCellComponent}
      />,
    );
    expect(toJson(tree)).toMatchSnapshot();
    expect(styleFn).toBeCalledWith({
      data,
      columns,
      row: data[1],
      rowIndex: 1,
    });
    expect(classNameFn).toBeCalledWith({
      data,
      columns,
      row: data[1],
      rowIndex: 1,
    });
  });
});
