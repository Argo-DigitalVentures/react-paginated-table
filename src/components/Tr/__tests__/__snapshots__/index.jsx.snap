exports[`<Tr /> should accept className and style functions 1`] = `
<tr
  className="test"
  style={
    Object {
      "color": "green",
    }
  }
  title="This is a row in the table.">
  <FakeCellComponent
    className=""
    column={
      Object {
        "header": "Company Name",
        "id": "company",
        "selector": "company",
      }
    }
    columnIndex={0}
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": "Location",
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "John Hancock",
        "location": "New York",
      }
    }
    rowIndex={1}
    style={Object {}} />
  <FakeCellComponent
    className=""
    column={
      Object {
        "header": "Location",
        "id": "location",
        "selector": "location",
      }
    }
    columnIndex={1}
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": "Location",
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "John Hancock",
        "location": "New York",
      }
    }
    rowIndex={1}
    style={Object {}} />
</tr>
`;

exports[`<Tr /> should accept className, style, and other DOM props 1`] = `
<tr
  className="test"
  style={
    Object {
      "color": "green",
    }
  }
  title="This is a row in the table.">
  <FakeCellComponent
    className=""
    column={
      Object {
        "header": "Company Name",
        "id": "company",
        "selector": "company",
      }
    }
    columnIndex={0}
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": "Location",
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "John Hancock",
        "location": "New York",
      }
    }
    rowIndex={1}
    style={Object {}} />
  <FakeCellComponent
    className=""
    column={
      Object {
        "header": "Location",
        "id": "location",
        "selector": "location",
      }
    }
    columnIndex={1}
    columns={
      Array [
        Object {
          "header": "Company Name",
          "id": "company",
          "selector": "company",
        },
        Object {
          "header": "Location",
          "id": "location",
          "selector": "location",
        },
      ]
    }
    data={
      Array [
        Object {
          "company": "Argo",
          "location": "New York",
        },
        Object {
          "company": "John Hancock",
          "location": "New York",
        },
      ]
    }
    row={
      Object {
        "company": "John Hancock",
        "location": "New York",
      }
    }
    rowIndex={1}
    style={Object {}} />
</tr>
`;
